### PR TITLE
feat: v0.4.0 — bug fixes, test coverage, features

### PR DESCRIPTION
**Pull request by apermo**
### Bug fixes
- #16 Gallery: missing line break after figure opening tag
- #17 Double `<p>` wrapping on styled paragraphs (new `unwrap_double_paragraphs` pipeline step)

### Features
- #20 Converter chaining: LIFO stack per tag, falls through on `can_convert()` rejection
- #15 New `wp classic-to-gutenberg detect` command — scans for unparseable content (table/csv/json output)

### Test coverage
- #25 Unit tests for MigrationRunner (8 tests)
- #22 Unit tests for ContentConverter (7 tests)
- #26 Unit tests for ClassicPostFinder (6 tests)
- #23 Unit tests for AdminNotice (7 tests)
- #24 Unit tests for RowAction (5 tests)
- #21 Integration test coverage uploaded to Codecov
- Shared `PostIdParserTrait` extracted for CLI commands

Total: 70 unit tests, 126 assertions (was 35 tests, 69 assertions)

### Remaining from milestone
- #9 Attribute/class allowlist — deferred to future release
- #28 Edge case fixtures — deferred to future release

## Test plan

- [x] PHPCS passes (0 errors, 0 warnings)
- [x] PHPStan passes
- [x] 70 unit tests pass
- [x] Integration tests pass (CI green)
- [x] E2E smoke tests pass (18/18 locally + CI)

### CODE REVIEW

**Comment by apermo** [comment]
### Code review

No issues found. Checked for bugs and CLAUDE.md compliance.

Self-reviewed: CLAUDE.md adherence, obvious bugs, git history context, previous PR comments, code comment compliance. Three potential issues were flagged and scored — all below threshold (10, 25, 25). CI passes (PHP CI, WP Integration, E2E). PR Validation failure is commit message format on individual commits — resolves on squash merge.

**Coverage:** 70 unit tests (126 assertions), up from 35 tests (69 assertions). Integration and E2E tests also pass.

File: src/CLI/DetectCommand.php [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+	private function find_classic_posts( array $assoc_args ): array {
+		$finder     = new ClassicPostFinder();
+		$query_args = [];
+
+		if ( isset( $assoc_args['post-type'] ) ) {
+			$query_args['post_type'] = \explode( ',', $assoc_args['post-type'] );
+		}
+
+		return $finder->find( $query_args );
+	}
```
[the remark this answers is not in the document]
Comment (apermo) [reply src/CLI/DetectCommand.php]: Prefer common abstract class if reasonable.

File: src/CLI/DetectCommand.php [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+	 *
+	 * @return string
+	 */
+	private function truncate_snippet( string $snippet ): string {
+		$snippet = \trim( $snippet );
+		$snippet = \preg_replace( '/\s+/', ' ', $snippet ) ?? $snippet;
+
+		if ( \strlen( $snippet ) > 80 ) {
+			return \substr( $snippet, 0, 77 ) . '...';
+		}
```
[the remark this answers is not in the document]
Comment (apermo) [reply src/CLI/DetectCommand.php:188]: Looks a bit overengineered here. I would consider simply documenting what 80 and 77 are.

File: src/Converter/BlockConverterFactory.php [hunk 3]
```diff
@@ -62,19 +75,28 @@ public function register( BlockConverterInterface $converter ): void {
...
 
-		if ( isset( $this->converters[ $tag ] ) && $this->converters[ $tag ]->can_convert( $tag, $html ) ) {
-			return $this->converters[ $tag ];
+		if ( isset( $this->converters[ $tag ] ) ) {
+			$stack = \array_reverse( $this->converters[ $tag ] );
+			foreach ( $stack as $converter ) {
+				if ( $converter->can_convert( $tag, $html ) ) {
+					return $converter;
+				}
+			}
```
[the remark this answers is not in the document]
Comment (apermo) [reply src/Converter/BlockConverterFactory.php:85]: we do have a plan of prioritizing the converters anyway, i think that still sits in the backlog. I assume that will be refatored anyway. Evaluate, if i'm right, add a //todo here to keep in mind but omit any changes. Update the issue accordingly.

File: src/ContentConverter.php [hunk 4]
```diff
@@ -140,6 +142,24 @@ private function restore_special_comments( string $content, array $placeholders
...
+	 * @param string $content The wpautop'd content.
+	 *
+	 * @return string
+	 */
+	private function unwrap_double_paragraphs( string $content ): string {
+		return (string) \preg_replace(
+			'/<p>\s*(<p\s[^>]*>.*?<\/p>)\s*<\/p>/s',
+			'$1',
+			$content,
+		);
```
[the remark this answers is not in the document]
Comment (apermo) [reply src/ContentConverter.php:160]: valid, but i would at least add error logging in that case.

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,25 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.4.0] - Unreleased
```
[the remark this answers is not in the document]
Comment (apermo) [reply CHANGELOG.md]: Will be taken care of on merge
